source/common/http/http2/codec_impl.cc

```diff
@@ -716,6 +716,16 @@ void ConnectionImpl::shutdownNotice() {
   }
 }
 
+void ConnectionImpl::protocolErrorForTest() {
+  int rc = nghttp2_submit_goaway(session_, NGHTTP2_FLAG_NONE,
+                                 nghttp2_session_get_last_proc_stream_id(session_),
+                                 NGHTTP2_PROTOCOL_ERROR, nullptr, 0);
+  ASSERT(rc == 0);
+
+  if (sendPendingFrames().ok()) {
+  }
```

Comment:
I would probably just return the error code here and assert it in the test itself.